docs/Blog-Post.md

```diff
@@ -0,0 +1,578 @@
+tensor parallism and FSDP. To us, this seems like a bit of a missed opportunity (relative to what we have in Haliax), but it's
+still useful.
+
+Haliax's NamedArrays are probably most similar to [Mesh-Tensorflow](https://github.com/tensorflow/mesh). PyTorch has
+[Named Tensors](https://pytorch.org/docs/stable/named_tensor.html). They're fairly new and "bolted on" to the existing
+positional tensors. They also don't help with model partitioning, which is one of the main use cases for named axes in
+Haliax.
+
+## Haliax Tutorials
+
+This is just a taste of what Haliax can do. For more details, please see our interactive tutorials on Colab:
+
+* [Introduction to Haliax with Transformers](https://colab.research.google.com/drive/1TiTcQQ4V5mopbgCu1SVl-oqJtXn7rFnC?usp=sharing)
+* [Scaling Transformers in Haliax](https://colab.research.google.com/drive/1QX4yH3zRFF3Xiibf1aahETcSQ5nbcUMz?usp=sharing), including FSDP in JAX.
+
+# Levanter: Bitwise Reproducible Foundation Models with JAX
+
+Levanter is a library for training foundation models built on top of Haliax. It provides a complete pipeline
+for training a GPT-2-like Transformer, complete with data preparation, logging, training, checkpointing, evaluation, and
+export, while maintaining bitwise reproducibility throughout.
+
+We have used Levanter to train models as large as 6.7b parameters on a v3-256, and have run experiments showing that it
+can scale up to least 65b parameters.
+
+## Bitwise Reproducibility
+
+One of the benefits of JAX is that it offers strong guarantees for reproducibility. In particular, JAX's fine-grained
+control over PRNG states makes it easy to ensure bitwise reproducibility, especially when using TPUs.
+Levanter takes advantage of this to offer bitwise reproducibility for training runs, even after preemption. In particular,
+the same run with the same code on the same set of hardware (e.g. a v3-32 or a v3-256) will produce the exact same loss curve, even if it is
+preempted and resumed multiple times. As an example, here is a screenshot of a training run being resumed multiple times, even on different TPU pod slices:
+
+![plot showing bitwise reproducibility with four training runs superimposed with the exact same loss curve](figures/bitwise_repro_curve.png)
+
+The fact that you can't make out the different lines is the point: the training runs are bitwise identical,
+a huge advantage for debugging and reproducibility.
+
+Levanter also logs everything necessary to exactly reproduce a run: the git SHA, code, configuration,
+and a pip-freeze of the environment. Checkpoints serialize the entire model state, including the optimizer state,
+as well as the "main" PRNG state, which is used to generate the other PRNG states. This means that you can
+exactly reproduce a run by simply checking out the git SHA, installing the dependencies, and running the code (on the same
+hardware configuration).
+
+## Efficiency and Scale
+
+XXX something something v3-256 scaling numbers?
```

Comment:
Make sure we transition: We talked about legibility, reproducibility.  Final property is scalability.